### ISSUE TITLE
Store server credentials securely

### ISSUE THREAD

**Issue by Krakonos** [openstreetmap/merkaartor#104]
They are currently stored in the configuration, as plain text. This has been like this for a long time and has to be fixed eventually. 

There is reasonably popular library frankosterfeld/qtkeychain that uses platform-specific password stores, and could be used.

Feel free to suggest other means/libraries. It would be great to do it with plain Qt without dragging more dependencies in, but I don't think Qt is capable of doing this.

**Comment by stephan75** [comment]
What about using OAuth?

There are several apps / software using it ... Like vespucci or Josm ...

**Comment by Krakonos** [comment]
Good idea. I actually have it on my notes somewhere. I'll keep it in mind as I refactor all the networking code.

**Comment by stephan75** [comment]
Sounds good!

I found some more details at http://wiki.openstreetmap.org/wiki/OAuth

**Comment by opk12** [comment]
Note that it will be OAuth 2 - OSM is going to drop OAuth 1 (#286). Then the access token would go in the keyring / password store.
